Make sure that color indices match MagicaVoxel tool

muit opened the pull request:
Hello,
I have been using your add-on for a while. It is truly useful!

Now, sadly I was having some trouble with textures.
Imported models did not have the same palette as the one seen on magica-voxel or the PNGs exported by that same tool.
This was because the indices displayed don't necessarily match those in the file.

See https://github.com/jpaver/opengametools/blob/0dce3f2277a32b9f646367139506b00da6f080d7/src/ogt_vox.h#L2198

I understand this may not be a desirable feature for your add-on, so I leave it as your choice if you want to merge it :)

PS: The VoxModel data layout has changed a bit to simplify working with colors. Also matching open game tools.

Thanks for the work on the add-on!
Muit

muit committed:
Make sure that color indices match MagicaVoxel tool

See https://github.com/jpaver/opengametools/blob/0dce3f2277a32b9f646367139506b00da6f080d7/src/ogt_vox.h#L2198

muit commented:
Hold on I found an small color artifact, fixing.

P.S. Fixed :)

muit committed:
Rotate palette by one on post-processing

muit commented:
As a comparison.
With this palette from magicavoxel
<img width="256" height="1" alt="Original Palette" src="https://github.com/user-attachments/assets/3097a534-c3c2-432e-957a-a65cf223c78f" />

I got before
<img width="256" height="1" alt="Palette before change" src="https://github.com/user-attachments/assets/f3b0379b-11cc-4639-a8df-d59549bf46b5" />

And now I get
<img width="256" height="1" alt="Current Palette" src="https://github.com/user-attachments/assets/3e6791ce-8a3e-43ba-a7f9-cda25a80cf4a" />

AstrorEnales commented:
Sorry for the late answer as i was on holiday. Thank you for your effort! I never really had a use-case for the index map, but you're right that it should be available. After a quick peek at the code it's looking good. I will test it thoroughly and merge it this or next week!

muit commented:
Lovely!

I still have my eye on that single pixel that gets rotated from the end to the start. From ogt that seems to be desirable, but its the only difference from the MG texture.

muit committed:
Mesh objects get their name from their node

If a node has one mesh, this mesh gets the same name as the node.

This is relevant when importing models into a game engine, where the mesh name and not the node is used for assets

muit committed:
Merge branch 'AstrorEnales:main' into main
